src/main/java/me/totalfreedom/totalfreedommod/discord/DiscordToAdminChatListener.java

```diff
@@ -38,57 +39,46 @@ public void onMessageReceived(MessageReceivedEvent event)
             String tag = dtml.getDisplay(member);
             StringBuilder message = new StringBuilder(ChatColor.DARK_GRAY + "[" + ChatColor.DARK_AQUA + "Discord" + ChatColor.DARK_GRAY + "] " + ChatColor.RESET);
             Message msg = event.getMessage();
-            for (Player player : Bukkit.getOnlinePlayers())
-            {
-                if (TotalFreedomMod.getPlugin().al.isAdmin(player))
-                {
-                    Admin admin = TotalFreedomMod.getPlugin().al.getAdmin(player);
-                    String format = admin.getAcFormat();
-                    if (format != null)
-                    {
-                        Displayable display = TotalFreedomMod.getPlugin().rm.getDisplay(player);
-                        net.md_5.bungee.api.ChatColor color = getColor(display);
-                        String m = format.replace("%name%", member.getEffectiveName())
-                                .replace("%rank%", getDisplay(member))
-                                .replace("%rankcolor%", color.toString())
-                                .replace("%msg%", FUtil.colorize(msg.getContentDisplay()));
-                        message.append(FUtil.colorize(m));
-                    }
-                    else
-                    {
-                        String m = ChatColor.DARK_RED + member.getEffectiveName() + " "
-                                + ChatColor.DARK_GRAY + tag + ChatColor.DARK_GRAY
-                                + ChatColor.WHITE + ": " + ChatColor.GOLD + FUtil.colorize(msg.getContentDisplay());
-                        message.append(m);
-                    }
-                }
-            }
 
-            ComponentBuilder builder = new ComponentBuilder(message.toString());
+            message.append(msg.getContentDisplay());
+
+            ComponentBuilder builder = new ComponentBuilder(msg.toString());
             if (!msg.getAttachments().isEmpty())
             {
                 for (Message.Attachment attachment : msg.getAttachments())
                 {
-                    if (attachment.getUrl() == null)
-                    {
-                        continue;
-                    }
-
                     TextComponent text = new TextComponent(ChatColor.YELLOW + "[Media]");
                     text.setClickEvent(new ClickEvent(ClickEvent.Action.OPEN_URL, attachment.getUrl()));
                     builder.append(text);
                     message.append("[Media]"); // for logging
                 }
             }
 
-            for (Player player : Bukkit.getOnlinePlayers())
+            FLog.info(message.toString());
+
+            Bukkit.getOnlinePlayers().stream().filter(player -> TotalFreedomMod.getPlugin().al.isAdmin(player)).forEach(player ->
```

Comment:
From here to line 81 should be before `[Media]` creation otherwise it will just produce a format something like `[Discord] [Media] [AC formatting]` - See https://github.com/AtlasMediaGroup/TotalFreedomMod/blob/5837b05a3569c2167180fb074069c25ca7d88956/src/main/java/me/totalfreedom/totalfreedommod/discord/DiscordToMinecraftListener.java for an example